cpu/stm32_common/periph/flashpage.c

```diff
@@ -65,6 +65,15 @@ static void _unlock_flash(void)
 #endif
 }
 
+static void _wait_for_pending_operations(void)
+{
+    DEBUG("[flashpage] waiting for any pending operation to finish\n");
+    while (FLASH->SR & FLASH_SR_BSY) {}
+    if (FLASH->SR & FLASH_SR_EOP) {
```

Comment:
The `if` section is not necessary to make it work on M3 nodes. I tested it.
So if needed it should go in its own commit with justification.

What makes it work is that now  the `while` is in a function so it is not getting re-ordered by gcc.
I think a comment on the function and a message in the commit is necessary.

Reply:
This is what is done in the STM32 Cube code, I just did the same. I have no idea if/when this may happen.

Reply:
Yes, then it should not be in a "fix M3" commit but in a "make it look more like stm32 cube code just in case".